cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1237,6 +1239,7 @@ func allocateMachine(ctx context.Context, logger *slog.Logger, ds *datastore.Ret
 		VPN:             allocationSpec.VPN,
 		FirewallRules:   firewallRules,
 		UUID:            uuid.New().String(),
+		DNSServers:      allocationSpec.DNSServers,
```

Comment:
some sort of validation is required like:
walk through all elements
if the element is a ip, validate with netip.ParseAddr()
else it must be a rfc conformant hostname

Reply:
This ensures a breaking OS installation